Workday/canvas-kit
# chore: Add custom lint rule to encourage CK slash imports

## Summary

Fixes: #1761

![category](https://img.shields.io/badge/release_category-Infrastructure-cornflowerblue)

---

## For the Reviewer

This PR:

- Adds a custom lint rule for check CK import statements for slash imports
  - This rule only applies to `canvas-kit-react`, `canvas-kit-preview-react`, and `canvas-kit-labs-react`
  - `canvas-kit-popup-stack`, `canvas-kit-codemod`, `canvas-kit-docs`, and `canvas-kit-css-*` are excluded
- Adds tests to verify the rule works
- Updates CK import statements that are not using slash imports

## Areas for Feedback?

I would like to know whether the error messages make sense, or if they should be reworded a bit.

## Where Should the Reviewer Start?

`utils/custom-lint-rules/use-ck-slash-imports.js`

## Testing Manually

The only way to test this manually would be:
-  pull the branch down
- update a CK import to use the base package
- run `yarn lint` and see the test fail

But the test I added is probably sufficient: `utils/custom-lint-rules/use-ck-slash-imports.spec.ts`. If it passes the lint rule works as expected.

## Thank You Gif (optional)

![a happy pack of dachshunds running through a field wearing hotdog costumes](https://media.giphy.com/media/NmGbJwLl7Y4lG/giphy.gif)
